core/templates/dev/head/pages/base.html

```diff
@@ -72,6 +72,8 @@
       {% include 'pages/header_css_libs.html' %}
     {% endblock header_css %}
 
+    <script src="{{TEMPLATE_DIR_PREFIX}}/pages/load_constants.js"></script>
```

Comment:
Yes, it's guaranteed. The `<script>` tags are loaded and run in the order they appear in HTML, this is first `<script>` tag in the page and Angular loads after. Also XMLHttpRequest that fetches the JSON is synchronous so it will block any loading of other scripts.